### PR TITLE
[DRFT-487] Clowderized system-baseline service

### PR DESCRIPTION
**Pull request by jpramos123**
## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [x] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: clowdapp.yaml [hunk 1]
```diff
@@ -0,0 +1,190 @@
+---
+
+apiVersion: v1
```
Comment (CodeHeeler) [inline clowdapp.yaml:3]: I'm a little confused by the two apiVersions present in this file, both v1 and v1alpha1.  Are you at times referring to like our views and openapi spec for the regular api and at other times referring to the new clowder app or should all of these be consistently one name (lines 3, 9, 18, 125, 145, 153)?

Also, if you have a cloudl.redhat.com/v1alpha1 endpoint, don't you need to also have views for this like we have for regular, internal, and mgmt already?

Comment (petrblaho) [reply clowdapp.yaml:3]: I am not sure but isn't this version of the clowder protocol? For the format of the content of this file? Just asking...

Comment (jpramos123) [reply clowdapp.yaml:3]: Hey @CodeHeeler, this is not related to our endpoints and services it self.

It's a specific Clowder configuration pattern that is specified in the docs (need to be in VPN):

https://clouddot.pages.redhat.com/clowder/dev/api_reference.html#k8s-api-github-com-redhatinsights-clowder-apis-cloud-redhat-com-v1alpha1-clowdapp

Here an example from the Clowder docs:

https://clouddot.pages.redhat.com/clowder/dev/usage/jobs.html

Thanks

File: run_app.sh [hunk 2]
```diff
@@ -1,22 +1,38 @@
-TEMPDIR=`mktemp -d`
+if [ -z "$ACG_CONFIG"]; then
```
Comment (petrblaho) [inline run_app.sh:1]: Does this come from clowder itself? Somehow automagically?

File: clowdapp.yaml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    value: 500m
+  - name: MEMORY_LIMIT
+    value: 500Mi
+  - name: CPU_REQUEST
+    value: 100m
+  - name: MEMORY_REQUEST
+    value: 300Mi
+  - description: ClowdEnv Name
+    name: ENV_NAME
+  - name: CLOWDER_ENABLED
```
Comment (petrblaho) [inline clowdapp.yaml:189]: Is this used somewhere? Or needed for clowder to work?

**Review by petrblaho — APPROVED** [review]
LGTM!

I would like my questions answered so I can learn something but if these are ok I am happy to ACK!

Edit: my questions were answered :-)

File: run_app.sh (hunk 2 above)
Comment (jpramos123) [reply run_app.sh:1]: @petrblaho Yes, this come from Clowder! This is inserted into the container so we can know if it's running with Clowder or not! Please look at:

https://clouddot.pages.redhat.com/clowder/dev/migration/migration.html#_code_changes_to_consume_configuration

Comment (petrblaho) [reply run_app.sh:1]: @jpramos123 thank you for your answer!

File: clowdapp.yaml (hunk 3 above)
Comment (jpramos123) [reply clowdapp.yaml:189]: ENV_NAME is automagically inserted by Clowder as well, so the clowdapp can reference the correct ClowdEnvironment.

Please, look at: https://clouddot.pages.redhat.com/clowder/dev/migration/migration.html#_create_deployment_template_with_clowdapp_resource

Comment (petrblaho) [reply clowdapp.yaml:189]: Awesome!

**Pull request merged**